nesta/packages/arxiv/tests/test_collect_arxiv.py

```diff
@@ -32,188 +31,190 @@ def mock_response():
+
+        _arxiv_request('another_test.url', delay=0, resumptionToken='123456')
+        assert mocked_requests.call_args[0] == ('another_test.url', )
+        assert mocked_requests.call_args[1] == {'params': {'verb': 'ListRecords', 'resumptionToken': '123456'}}
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv.requests.get')
+    def test_arxiv_request_returns_xml_element(self, mocked_requests, mock_response):
+        mocked_requests().text = mock_response
+        response = _arxiv_request('test.url', delay=0, metadataPrefix='arXiv')
+        assert type(response) == ET.Element
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv.requests.get')
+    def test_arxiv_request_implements_delay(self, mocked_requests, mock_response):
+        """Fair use policy specifies wait of 3 seconds between each request:
+            https://arxiv.org/help/api/user-manual#Quickstart
+        """
+        mocked_requests().text = mock_response
+
+        start_time = time.time()
+        _arxiv_request('test.url', delay=1, metadataPrefix='arXiv')
+        assert time.time() - start_time > 1
+
+        start_time = time.time()
+        _arxiv_request('test.url', delay=3, metadataPrefix='arXiv')
+        assert time.time() - start_time > 3
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_request_token_returns_correct_token(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        assert request_token() == '3132962'
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_request_token_doesnt_override_delay(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        request_token()
+        if 'delay' in mocked_request.call_args[1]:
+            assert mocked_request.call_args[1]['delay'] >= 3
+
+
+class TestTotalArticles:
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_total_articles_returns_correct_amount(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        assert total_articles() == 1463679
+
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_total_articles_doesnt_override_delay(self, mocked_request, mock_response):
+        mocked_request.return_value = ET.fromstring(mock_response)
+        total_articles()
+        if 'delay' in mocked_request.call_args[1]:
+            assert mocked_request.call_args[1]['delay'] >= 3
+
+
+class TestArxivBatch:
+    @mock.patch('nesta.packages.arxiv.collect_arxiv._arxiv_request', autospec=True)
+    def test_arxiv_batch_sends_resumption_token_if_provided(self, mocked_request):
+        batch, _ = arxiv_batch('111222444|1')
+        assert mocked_request.mock_calls[0] == mock.call('http://export.arxiv.org/oai2',
```

Comment:
there are a large number of repeated variables like `http://export.arxiv.org/oai2` and `111222444|1`. Might be an idea to hard code them to some `UPPER_CASE` global (or fixtures, but probs overkill) at the top of the script. (Probably could do this for the `mock.patch` args if they're repeated often

Reply:
this is done, looks much better 👍 thanks!